### PR TITLE
#49380: Introduce `is_post_status_viewable()` and `is_post_viewable()` functions.

### PR DESCRIPTION
**Pull request by peterwilsoncc**
https://core.trac.wordpress.org/ticket/49380

### CODE REVIEW

File: src/wp-includes/post.php [hunk 1]
```diff
@@ -1987,6 +1987,48 @@ function is_post_type_viewable( $post_type ) {
 	return $post_type->publicly_queryable || ( $post_type->_builtin && $post_type->public );
 }
 
+/**
+ * Determines whether a post status is considered "viewable".
+ *
+ * For built-in post types, the 'public' value will be evaluated.
+ * For all others, the 'publicly_queryable' value will be used.
+ *
+ * @since x.x.x
```
Comment (peterwilsoncc) [inline src/wp-includes/post.php:1996]: :1234: `@TODO @since`s

File: src/wp-includes/post.php [hunk 2]
```diff
@@ -1987,6 +1987,48 @@ function is_post_type_viewable( $post_type ) {
...
+ */
+function is_post_status_viewable( $post_status ) {
+	if ( is_scalar( $post_status ) ) {
+		$post_status = get_post_status_object( $post_status );
+		if ( ! $post_status ) {
+			return false;
+		}
+	}
+
+	return $post_status->publicly_queryable || ( $post_status->_builtin && $post_status->public );
```
Comment (peterwilsoncc) [inline src/wp-includes/post.php:2009]: :1234: Add filters per https://core.trac.wordpress.org/ticket/49628

File: src/wp-includes/capabilities.php [hunk 3]
```diff
@@ -223,6 +223,16 @@ function map_meta_cap( $cap, $user_id, ...$args ) {
 				}
 			}
 
+			if ( is_post_viewable( $post ) ) {
```
Comment (peterwilsoncc) [inline src/wp-includes/capabilities.php:226]: ⚠️ This may not be a sensible thing to do. Tests would need to change if added.

Comment (peterwilsoncc) [reply src/wp-includes/capabilities.php:226]: TODOs:

* [ ] Move below `( ! $post_type->map_meta_cap )` check

Comment (peterwilsoncc) [reply src/wp-includes/capabilities.php]: ^ Reverted experiment in 19588eb7a4968fa2210ca534935bbc6b0d263e9d

**Comment by whyisjake** [comment]
> Words.

**Pull request closed without merging**